zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -20,9 +30,108 @@ impl std::ops::DerefMut for TransactionRecordsById {
         &mut self.0
     }
 }
+
+/// This block implements constructors.
+impl Default for TransactionRecordsById {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl TransactionRecordsById {
+    // New empty Map
+    pub fn new() -> Self {
+        TransactionRecordsById(HashMap::new())
+    }
     // Associated function to create a TransactionRecordMap from a HashMap
     pub fn from_map(map: HashMap<TxId, TransactionRecord>) -> Self {
         TransactionRecordsById(map)
     }
 }
+/// This block implements methods to modify the map.
+impl TransactionRecordsById {
+    /// All information after a certain height is invalidated during a reorg.
+    pub fn invalidate_all_transactions_after_or_at_height(&mut self, reorg_height: BlockHeight) {
```

Comment:
does this function need such a cumbersome name?
```suggestion
    pub fn truncate_transactions(&mut self, reorg_height: BlockHeight) {
```

Reply:
yes